src/generators/operationGenerator.ts

```diff
@@ -958,7 +958,7 @@ function hasMediaType(
   operationDetails: OperationDetails,
   mediaType: KnownMediaType
 ) {
-  if (!operationDetails.requests) {
+  if (!operationDetails.requests.some(r => !!r.mediaType)) {
```

Comment:
requests is never undefined so this check was never true. Checking for the mediaType field inside request instead.